cluster-api/gcp-deployer/machine_setup_configs.yaml

```diff
@@ -0,0 +1,379 @@
+items:
```

Comment:
I'm curious why this file isn't defined as a ConfigMap but just a YAML list of things? To someone just browsing the code, it would not be obvious that these key=value pairs are consumed as a ConfigMap is the cluster.

Reply:
> We don't want to work with the actual configmap API object in case the machine controller is ever run on a non-kubernetes cluster

Are we supporting that use case? When can the controller run if not in a k8s cluster (even bootstrapping will happen in a K8s cluster.

> it's easier to just work with the single file.

That's fine. We can have everything in one file still. That's not my concern. My concern is that we are hiding this from the user and it's not obvious to them how to update these values. If this was using the ConfigMap object itself, they could simply kubectl apply it rather than mucking with documentation.

So to summarize - I'm not saying we need to split things, but rather the abstraction of ConfigMap be pulled up in this spec file rather than deep in code somewhere.

Reply:
> When can the controller run if not in a k8s cluster (even bootstrapping will happen in a K8s cluster.

Right now the gcp deployer is bootstrapping outside of a k8s cluster. In the future when we have real bootstrapping we can switch over to setting up a configmap from the start, but the machinesetup library is set up to handle a yaml file. If the APIServer ever goes down, we don't want to rely on the configmap object and instead work with the volume mounted yaml file.

Changing the yaml to a configmap definition would add a lot of extra logic to the gcp deployer since it would have to serialize and write out the contents to a file so the library can consume it. The gcp deployer is also just temporary until we have actual bootstrapping set up, so I don't think it's worth adding all the extra logic when it'll be replaced anyway. 

> it's not obvious to them how to update these values. If this was using the ConfigMap object itself, they could simply kubectl apply it rather than mucking with documentation.

Once the cluster is created and running, they can still use kubectl apply to update the configmap. If they want to edit the values before creating the cluster, they'd edit the yaml file the same way they can edit the machines or cluster yaml. Again, we can move to transparently setting up configmap in the future with real bootstrapping and it would be more obvious how to update, but I think for now just documenting what I said above would be enough.

Reply:
I disagree about the extra logic concern you have. Currently the matching logic and the mounting logic are complex and opaque to the user. If anything, it will simplify the configmap a whole lot more. But I don't want to block this PR on that.

Please add your explanation as a comment and a TODO with an issue as a future improvement.